crates/rattler_package_streaming/src/fs.rs

```diff
@@ -0,0 +1,51 @@
+use crate::{ArchiveType, ExtractError};
+use std::fs::File;
+use std::path::Path;
+
+/// Extracts the contents a `.tar.bz2` package archive from the specified remote location.
+///
+/// ```rust,no_run
+/// # use std::path::Path;
+/// use rattler_package_streaming::fs::extract_tar_bz2;
+/// let _ = extract_tar_bz2(
+///     Path::new("conda-forge/win-64/python-3.11.0-hcf16a7b_0_cpython.tar.bz2"),
+///     Path::new("/tmp"))
+///     .unwrap();
+/// ```
+pub fn extract_tar_bz2(archive: &Path, destination: &Path) -> Result<(), ExtractError> {
+    let file = File::open(archive)?;
+    crate::read::extract_tar_bz2(file, destination)
+}
+
+/// Extracts the contents a `.conda` package archive at the specified path.
+///
+/// ```rust,no_run
+/// # use std::path::Path;
+/// use rattler_package_streaming::fs::extract_conda;
+/// let _ = extract_conda(
+///     Path::new("conda-forge/win-64/python-3.11.0-hcf16a7b_0_cpython.conda"),
+///     Path::new("/tmp"))
+///     .unwrap();
+/// ```
+pub fn extract_conda(archive: &Path, destination: &Path) -> Result<(), ExtractError> {
+    let file = File::open(archive)?;
+    crate::read::extract_conda(file, destination)
+}
+
+/// Extracts the contents a package archive from the specified remote location. The type of package
+/// is determined based on the path of the url.
```

Comment:
based on the file extension of the path?

Reply:
Changed